esp/services/ws_dfu/ws_dfuService.cpp

```diff
@@ -5885,4 +5885,182 @@ int CWsDfuEx::GetIndexData(IEspContext &context, bool bSchemaOnly, const char* i
+
+    fileDesc->setDefaultDir(curDir.str());
+
+    StringBuffer partMask;
+    unsigned parts = group->ordinality();
+    getPartMask(partMask, logicalName, parts);
+    fileDesc->setNumParts(parts);
+    fileDesc->setPartMask(partMask);
+
+    ClusterPartDiskMapSpec mspec;
+    mspec.defaultCopies = DFD_DefaultCopies;
+    fileDesc->addCluster(groupName, group, mspec);
+
+    SecAccessFlags perm = queryDistributedFileDirectory().getFDescPermissions(fileDesc, userDesc, DALI_LDAP_AUDIT_REPORT | DALI_LDAP_READ_WANTED | DALI_LDAP_WRITE_WANTED);
+    if (!HASWRITEPERMISSION(perm))
+        throw MakeStringException(ECLWATCH_INVALID_INPUT, "Failed to create %s. Permission denied.", logicalName);
+
+    //Publish the file
+    Owned<IDistributedFile> file = queryDistributedFileDirectory().createNew(fileDesc);
+    file->setAccessed();
+    file->setECL(recordDefinition);
+    file->attach(logicalName, userDesc);
+}
+
+void CWsDfuEx::getFilePartsOnCluster(IEspContext &context, const char *logicalName, const char *groupName,
+    IUserDescriptor *userDesc, IEspDFUFilePartsOnCluster &partsOnCluster)
+{
+    Owned<IDistributedFile> file = queryDistributedFileDirectory().lookup(logicalName, userDesc);
+    if(!file)
+        throw MakeStringException(ECLWATCH_FILE_NOT_EXIST,"Cannot find file %s.", logicalName);
+
+    StringBuffer clusterName;
+    Owned<IFileDescriptor> fileDesc;
+    unsigned numClusters = file->numClusters();
+    for (unsigned i = 0; i < numClusters; i++)
+    {
+        StringBuffer gName;
+        file->getClusterGroupName(i, gName);
+        if (strieq(groupName, gName.str()))
+        {
+            file->getClusterName(i, clusterName);
+            if(!clusterName.isEmpty()) //Should not be empty. But, check it just in case.
+                fileDesc.setown(file->getFileDescriptor(clusterName.str()));
+            break;
+        }
+    }
+    if(clusterName.isEmpty())
+        throw MakeStringException(ECLWATCH_FILE_NOT_EXIST,"Cannot find %s on any cluster.", logicalName);
+    if(!fileDesc)
+        throw MakeStringException(ECLWATCH_FILE_NOT_EXIST,"Cannot find FileDescriptor for %s on cluster %s.", logicalName, clusterName.str());
+
+    IArrayOf<IConstDFUPart> &filePartList = partsOnCluster.getDFUFileParts();
+    Owned<IPartDescriptorIterator> pi = fileDesc->getIterator();
+    ForEach(*pi)
+    {
+        IPartDescriptor &part = pi->query();
+
+        __int64 size = -1;
+        IPropertyTree* partPropertyTree = &part.queryProperties();
+        if (partPropertyTree)
+            size = partPropertyTree->getPropInt64("@size", -1);
+
+        for (unsigned int i=0; i<part.numCopies(); i++)
+        {
+            StringBuffer b;
+            part.queryNode(i)->endpoint().getUrlStr(b);
+
+            Owned<IEspDFUPart> filePart = createDFUPart("","");
+            filePart->setId(1 + part.queryPartIndex());
+            filePart->setPartSizeInt64(size);
```

Comment:
is there a non Int64 partsize we need to set as well?

Reply:
not new, but including the type in the method name seems a little odd, i.e. why is it not called setPartSize